shapeshift/web
# feat: display toast when a keepkey update is available

## Description
We need to display a toast in case there are some bootloader and fimware update available when an user connect a KeepKey.

In order to achieve this:
- I move the request logic into a reusable function used by both the `hook` and the `KeepKeyProvider`
- I added a toast logic to the KeepKeyProvider



## Notice




## Pull Request Type


## Issue (if applicable)


closes #1599
## Risk



## Testing
- Update your env var url as in the file changes
- Try to connect using a non up-to-date KeepKey, you should see one toast
- You can dismiss the toast, but in case you connect the keepkey again, it should be shown
- If it didn't get dismissed, you shouldn't see more than one toast

You may encounter CORS issues while testing on localhost


## Screenshots (if applicable)
![image](https://user-images.githubusercontent.com/14963751/167463519-f81d0f27-e7f9-42f2-99d5-28e827eebe22.png)
